[SwiftSyntax] Avoid linking clangAST in SwiftSyntaxParserLib build

ahoppen opened the pull request:
`printClangDeclName` introduced another use of the clang AST in `swiftAST`, remove it by returning early if we are only building the parser library.

ahoppen commented:
@swift-ci Please smoke test

akyrtzi commented:
Could we avoid linking `clangAST` by avoiding the call via `#if SWIFT_BUILD_ONLY_SYNTAXPARSERLIB`?
See an example with this commit: https://github.com/apple/swift/commit/5c956ba2e147e309da7c4e619587adade1584b7c

ahoppen committed:
[SwiftSyntax] Avoid linking clangAST in SwiftSyntaxParserLib build

`printClangDeclName` introduced another use of the clang AST in `swiftAST`, remove it by returning early if we are only building the parser library.

ahoppen commented:
Yes, that’s definitely the cleaner solution.

ahoppen commented:
@swift-ci Please smoke tet

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please smoke test Windows

akyrtzi commented:
Windows failure is unrelated.
